packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -619,4 +631,26 @@ export class Cluster extends ClusterBase {
       return nodeCount;
     }
   }
+
+  /**
+   * Adds a parameter to the Clusters' parameter group
+   *
+   * @param name the parameter name
+   * @param value the parameter name
+   */
+  public addToParameterGroup(name: string, value: string): void {
+    if (!this.parameterGroup) {
+      const param: { [name: string]: string } = {};
+      param[name] = value;
+      this.parameterGroup = new ClusterParameterGroup(this, 'ParameterGroup', {
+        description: `Parameter Group for the ${this.cluster.clusterIdentifier ?? Names.uniqueResourceName(this, {})} Redshift cluster`,
```

Comment:
The [clusterIdentifier](https://github.com/dontirun/aws-cdk/blob/2191f587dcdddf40df0ee68eab92c9b6eb606964/packages/%40aws-cdk/aws-redshift/lib/cluster.ts#L528) can be undefined.  It's being set to the optional [clusterName prop](https://github.com/dontirun/aws-cdk/blob/2191f587dcdddf40df0ee68eab92c9b6eb606964/packages/%40aws-cdk/aws-redshift/lib/cluster.ts#L203)

Reply:
Agreed with the false information. 

It looks like the [cluster Identifier](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-redshift-cluster.html#cfn-redshift-cluster-clusteridentifier) needs to be unique account wide. So instead of setting the name to with `Names.uniqueResourceName` I think it might be better to fall back to [the  paramter group default description](https://github.com/dontirun/aws-cdk/blob/dynamic-redshift-parameter-groups/packages/%40aws-cdk/aws-redshift/lib/parameter-group.ts#L80) if the identifier isn't provided

Reply:
Removed the `Names.uniqueResourceName` and [falling back to the default](https://github.com/aws/aws-cdk/pull/20944/files#diff-1ae118454a3ef75fce1f5c54ea531575d0b77d43bb88b914898ccd93aac3c15aR645)